tests/test_util/test_warning.py

```diff
@@ -224,3 +224,22 @@ def test_method_deprecated_method_pickling(recwarn):
 def test_deprecated_callable_resolves_type(recwarn, func, _type):
     func(2)
     assert _type in recwarn.list[0].message.args[0]
+
+
+@pytest.mark.filterwarnings("ignore::DeprecationWarning")
+def test_function_deprecated_args_deprecated_callable_chained_decorators(recwarn):
+    @deprecated_args("2023.6", "x is not descriptive", [("x", "a")])
+    @deprecated_args("2023.6", "y is not descriptive", [("y", "b")])
+    @deprecated_callable(
```

Comment:
This does not match the signature of `deprecated_callable`. The old function name is deduced from what is decorated and I think while it's possible to overload the usage of one variable to be either the new name or, in it's absence, implying it is being discontinued. What if the developer forgets to add the new name, but `is_discontinued==False` an exception should be raised (it's not currently, but your approach here makes me realise it should be).

I think the approach to using one argument in `deprecated_args()` as `old_to_new_or_discontinued` relies on implicit logic because one could have a mix of types on the same iterable (e.g. `[('a', 'x'),'drop_this',('b', 'y')]`. It can be done, but to me this is less clear than
```python
@deprecated_callable(old_new=[('a', 'x'), `('b', 'y')], discontinued='drop_this')`
```
So, parroting the zen ...explicit is better than implicit.

Reply:
That was a lot more fiddly than it should have been.  But it is now implemented as you expect.  It's now so explict that even duck typing can't be used to discriminate between old_new and discontinued.  Argument keywords must be used if discontinued is present in the arguments.